middle_end/flambda2/to_cmm/un_cps_closure.ml

```diff
@@ -440,23 +440,28 @@ module Greedy = struct
   let rec create_closure_slots set state exported_code = function
     | [] -> state
     | (c, code_id) :: r ->
-        let s, state =
+        let slot =
           match find_closure_slot state c with
-          | Some s -> s, state
+          | Some s -> Some (s, state)
           | None ->
-              let calling_convention =
-                Exported_code.find_calling_convention exported_code code_id
-              in
+            match
+              Exported_code.find_calling_convention exported_code code_id
+            with
+            | Deleted -> None
```

Comment:
Why can `un_cps_closure` see (or rather has to deal with) closures in sets that have been deleted, whereas `un_cps` does not see these ?
I.e. the closure_id map that `un_cps` uses only contains the closures that have not been deleted (since the `un_cps` code uses a `map is empty` ~ `has zero closures` equivalence), and `un_cps_closure` uses basically the same map (as a list), so, either we should have a fatal_error here, or the code to detect the zero closure case in `un_cps` is wrong, or I'm missing something.